src/dataflow-types/src/client/controller.rs

```diff
@@ -277,6 +280,7 @@ where
         let replica_name = generate_replica_service_name(instance_id, replica_id);
         if let ConcreteComputeInstanceReplicaConfig::Managed {
             size_config: _size_config,
+            availability_zone: _az,
         } = config
```

Comment:
Nit: This could be rewritten to:
```
if matches!(config, ConcretComputeInstanceReplicaConfig::Managed { .. }) {
```

Reply:
Ah--I did this on purpose to force a full expansion of the enum's fields. I think this behavior/invariant isn't very obvious, so want to make sure any additions to this enum in the future trigger a compilation error to suggest to future devs that they should consider any impact the config might have on maintaining the service.